exasol/ds/sandbox/lib/run_create_vm.py

```diff
@@ -1,39 +1,47 @@
+from exasol.ds.sandbox.lib.ansible.ansible_repository import (
+    AnsibleRepository, default_repositories)
+from exasol.ds.sandbox.lib.ansible.ansible_run_context import (
+    AnsibleRunContext, default_ansible_run_context,
+    reset_password_ansible_run_context)
 from exasol.ds.sandbox.lib.asset_id import AssetId
 from exasol.ds.sandbox.lib.aws_access.aws_access import AwsAccess
 from exasol.ds.sandbox.lib.config import ConfigObject
-from exasol.ds.sandbox.lib.export_vm.run_make_ami_public import run_make_ami_public
-from exasol.ds.sandbox.lib.logging import get_status_logger, LogType
-
-from exasol.ds.sandbox.lib.setup_ec2.host_info import HostInfo
 from exasol.ds.sandbox.lib.export_vm.run_export_vm import export_vm
-from exasol.ds.sandbox.lib.setup_ec2.run_install_dependencies import run_install_dependencies
-from exasol.ds.sandbox.lib.setup_ec2.run_reset_password import run_reset_password
-from exasol.ds.sandbox.lib.setup_ec2.run_setup_ec2 import run_lifecycle_for_ec2, \
-    EC2StackLifecycleContextManager
+from exasol.ds.sandbox.lib.export_vm.run_make_ami_public import \
+    run_make_ami_public
+from exasol.ds.sandbox.lib.logging import LogType, get_status_logger
+from exasol.ds.sandbox.lib.setup_ec2.host_info import HostInfo
+from exasol.ds.sandbox.lib.setup_ec2.run_install_dependencies import \
+    run_install_dependencies
+from exasol.ds.sandbox.lib.setup_ec2.run_reset_password import \
+    run_reset_password
+from exasol.ds.sandbox.lib.setup_ec2.run_setup_ec2 import (
+    EC2StackLifecycleContextManager, run_lifecycle_for_ec2)
 from exasol.ds.sandbox.lib.setup_ec2.source_ami import find_source_ami
 
-
 LOG = get_status_logger(LogType.CREATE_VM)
 
 
-def run_create_vm(aws_access: AwsAccess, ec2_key_file: Optional[str], ec2_key_name: Optional[str],
-                  ansible_access: AnsibleAccess, default_password: str,
-                  vm_image_formats: Tuple[str, ...],
-                  asset_id: AssetId,
-                  configuration: ConfigObject,
-                  user_name: Optional[str],
-                  make_ami_public: bool,
-                  ansible_run_context=default_ansible_run_context,
-                  ansible_reset_password_context=reset_password_ansible_run_context,
-                  ansible_repositories: Tuple[AnsibleRepository, ...] = default_repositories) -> None:
+def run_create_vm(
+    aws_access: AwsAccess,
+    ec2_key_file: Optional[str],
+    ec2_key_name: Optional[str],
+    ansible_access: AnsibleAccess,
+    default_password: str,
+    vm_image_formats: Tuple[str, ...],
+    asset_id: AssetId,
+    configuration: ConfigObject,
+    user_name: Optional[str],
+    make_ami_public: bool,
+    ansible_run_context: AnsibleRunContext = default_ansible_run_context,
+    ansible_reset_password_context: AnsibleRunContext = reset_password_ansible_run_context,
+    ansible_repositories: Tuple[AnsibleRepository, ...] = default_repositories,
+    ec2_instance_type: str = "t2.medium",
```

Comment:
Does a default  still make sense here?

Reply:
I am happy to remove it. I was already tempted. 
Actually, we could agree that only the command line should define defaults.

But making the parameter mandatory requires moving it before the optional parameters  `ansible_...`.
I need to check the callers then. 
On the other hand, I already took care, to use `kwargs` in all places currently affected. 
Maybe this is a good opportunity?

Reply:
I have removed the default value for parameter `ec2_instance_type` in API levels below CLI.
Please see latest push.